src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -198,6 +198,18 @@ bool concat_in_place_optimization::match(const program_node& concat_node,
         }
         auto input_padd = pred.first->get_output_layout().data_padding;
 
+        // If static padding exists in non dyn_pad axis, returns false to avoid runtime optimized out.
+        if (is_runtime) {
+            auto is_dynamic_pad = input_padd.get_dynamic_pad_dims().sizes(format::get_default_format(layout::max_rank()));
+            for (size_t j = 0; j < layout::max_rank(); j++) {
+                if (is_dynamic_pad[j] == 0) {
+                    if ((concat_params.get_output_layout().data_padding.lower_size().sizes(def_fmt)[j] != 0)
+                        || (concat_params.get_output_layout().data_padding.upper_size().sizes(def_fmt)[j] != 0))
+                        return false;
+                }
+            }
+        }
+
```

Comment:
I think we dont need to check pred's dynamic padd.  We just need to check at runtime, 
1) if there is any other static pad dim in concat, which is not concat_axis. 
2) Also now that it does not use pred layout info, I think we can pull this change up, because this does not need pred info. (i.e., before line 84)

Reply:
Updated! Thanks!